<a href="https://colab.research.google.com/github/Coobaltoo13/Challenge-Amigo-Secreto/blob/main/Desafio1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [11]:
import pandas as pd

df = pd.read_csv("tienda_1 .csv")
df2 = pd.read_csv("tienda_2.csv")
df3 = pd.read_csv("tienda_3.csv")
df4 = pd.read_csv("tienda_4.csv")

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

In [ ]:
from google.colab import drive
drive.mount('/content/drive')


In [ ]:
!ls


 sample_data  'tienda_1 .csv'   tienda_2.csv   tienda_3.csv   tienda_4.csv


In [ ]:
df

Producto Categoría del Producto     Precio  Costo de envío  \
0     Asistente virtual           Electrónicos   164300.0          6900.0   
1       Mesa de comedor                Muebles   192300.0          8400.0   
2         Juego de mesa               Juguetes   209600.0         15900.0   
3            Microondas      Electrodomésticos   757500.0         41000.0   
4      Silla de oficina                Muebles   335200.0         20200.0   
...                 ...                    ...        ...             ...   
2354          Iphone 15           Electrónicos  1284400.0         66600.0   
2355          Impresora           Electrónicos   282800.0         15300.0   
2356      Juego de mesa               Juguetes   122900.0          7200.0   
2357           Cama box                Muebles   691300.0         35000.0   
2358         Smartwatch           Electrónicos   195000.0          8600.0   

     Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0         16/01/2021      Pedro Gomez          Bogotá             4   
1         18/05/2022  Beatriz Morales        Medellín             1   
2         15/03/2021   Juan Fernandez       Cartagena             1   
3         03/05/2022   Juan Fernandez            Cali             4   
4         07/11/2020    Maria Alfonso        Medellín             5   
...              ...              ...             ...           ...   
2354      15/11/2021  Izabela de León        Medellín             1   
2355      18/03/2023    Lucas Olivera          Bogotá             5   
2356      06/07/2021    Lucas Olivera            Cali             5   
2357      02/05/2022      Pedro Gomez           Pasto             1   
2358      18/06/2022    Felipe Santos            Cali             4   

          Método de pago  Cantidad de cuotas       lat       lon  
0     Tarjeta de crédito                   8   4.60971 -74.08175  
1     Tarjeta de crédito                   4   6.25184 -75.56359  
2     Tarjeta de crédito                   1  10.39972 -75.51444  
3                  Nequi                   1   3.43722 -76.52250  
4                  Nequi                   1   6.25184 -75.56359  
...                  ...                 ...       ...       ...  
2354               Nequi                   1   6.25184 -75.56359  
2355               Nequi                   1   4.60971 -74.08175  
2356  Tarjeta de crédito                   4   3.43722 -76.52250  
2357  Tarjeta de crédito                   2   1.21361 -77.28111  
2358  Tarjeta de crédito                   1   3.43722 -76.52250  

[2359 rows x 12 columns]

In [ ]:
df2

Producto   Categoría del Producto     Precio  Costo de envío  \
0     Silla de oficina                  Muebles   283600.0         15300.0   
1            Pandereta   Instrumentos musicales    76800.0          2300.0   
2            Cama king                  Muebles   443300.0         23800.0   
3            Cama king                  Muebles   906200.0         46400.0   
4     Silla de oficina                  Muebles   397800.0         19400.0   
...                ...                      ...        ...             ...   
2354      Set de ollas  Artículos para el hogar   140300.0          8100.0   
2355  Lavadora de ropa        Electrodomésticos  1220100.0         65200.0   
2356        Microondas        Electrodomésticos   797900.0         42900.0   
2357         Iphone 15             Electrónicos  2224300.0        116600.0   
2358         Bicicleta     Deportes y diversión   551000.0         27500.0   

     Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0         04/08/2021  Izabela de León        Medellín             1   
1         26/11/2020   Blanca Ramirez        Medellín             3   
2         22/10/2021  Izabela de León         Pereira             5   
3         04/05/2022    Maria Alfonso            Cali             5   
4         26/02/2022    Lucas Olivera          Bogotá             4   
...              ...              ...             ...           ...   
2354      04/12/2022  Ángel Rodriguez    Barranquilla             5   
2355      09/02/2020   Santiago Silva          Bogotá             4   
2356      27/09/2022   Blanca Ramirez          Bogotá             5   
2357      21/04/2022   Juan Fernandez        Medellín             5   
2358      03/09/2022    Maria Alfonso        Medellín             4   

          Método de pago  Cantidad de cuotas       lat       lon  
0                  Nequi                   1   6.25184 -75.56359  
1     Tarjeta de crédito                   4   6.25184 -75.56359  
2     Tarjeta de crédito                  10   4.81333 -75.69611  
3     Tarjeta de crédito                  10   3.43722 -76.52250  
4     Tarjeta de crédito                   3   4.60971 -74.08175  
...                  ...                 ...       ...       ...  
2354  Tarjeta de crédito                   2  10.96854 -74.78132  
2355  Tarjeta de crédito                   5   4.60971 -74.08175  
2356               Nequi                   1   4.60971 -74.08175  
2357  Tarjeta de crédito                   2   6.25184 -75.56359  
2358  Tarjeta de crédito                   1   6.25184 -75.56359  

[2359 rows x 12 columns]

In [ ]:
df3


Producto  Categoría del Producto     Precio  \
0            Secadora de ropa       Electrodomésticos  1285100.0   
1               Mesa de noche                 Muebles   290100.0   
2           Guitarra acústica  Instrumentos musicales   303500.0   
3           Balón de voleibol    Deportes y diversión    58800.0   
4                     Armario                 Muebles   521100.0   
...                       ...                     ...        ...   
2354  Dashboards con Power BI                  Libros    26700.0   
2355  Dashboards con Power BI                  Libros    31800.0   
2356         Silla de oficina                 Muebles   477100.0   
2357           Mesa de centro                 Muebles   123100.0   
2358      Modelado predictivo                  Libros    63400.0   

      Costo de envío Fecha de Compra         Vendedor Lugar de Compra  \
0            66600.0      17/12/2021    Camila Rivera     Santa Marta   
1            13600.0      10/02/2021    Maria Alfonso            Cali   
2            17400.0      22/01/2023    Lucas Olivera          Bogotá   
3             3300.0      22/03/2022      Pedro Gomez    Barranquilla   
4            25900.0      12/01/2022   Blanca Ramirez          Bogotá   
...              ...             ...              ...             ...   
2354          2100.0      21/10/2022  Ángel Rodriguez          Bogotá   
2355          2200.0      23/06/2022   Blanca Ramirez         Pereira   
2356         25600.0      19/02/2023    Camila Rivera       Cartagena   
2357          4700.0      19/09/2021   Blanca Ramirez          Bogotá   
2358          9400.0      24/05/2022  Mariana Herrera        Medellín   

      Calificación      Método de pago  Cantidad de cuotas       lat       lon  
0                4  Tarjeta de crédito                   4  11.24079 -74.19904  
1                3  Tarjeta de crédito                   1   3.43722 -76.52250  
2                5  Tarjeta de crédito                   3   4.60971 -74.08175  
3                4  Tarjeta de crédito                   6  10.96854 -74.78132  
4                1  Tarjeta de crédito                   1   4.60971 -74.08175  
...            ...                 ...                 ...       ...       ...  
2354             5  Tarjeta de crédito                   1   4.60971 -74.08175  
2355             5  Tarjeta de crédito                   1   4.81333 -75.69611  
2356             5  Tarjeta de crédito                   3  10.39972 -75.51444  
2357             4  Tarjeta de crédito                   8   4.60971 -74.08175  
2358             5  Tarjeta de crédito                  10   6.25184 -75.56359  

[2359 rows x 12 columns]

In [ ]:
df4

Producto   Categoría del Producto     Precio  Costo de envío  \
0            Lavavajillas        Electrodomésticos  1084800.0         56000.0   
1                Cama box                  Muebles   473800.0         25400.0   
2     Balón de baloncesto     Deportes y diversión    23700.0             0.0   
3     Modelado predictivo                   Libros    57700.0          1300.0   
4             Auriculares             Electrónicos   138900.0          5600.0   
...                   ...                      ...        ...             ...   
2353      Mesa de comedor                  Muebles   227700.0         12600.0   
2354     Silla de oficina                  Muebles   342000.0         18600.0   
2355         Vaso térmico     Deportes y diversión    36700.0          2400.0   
2356         Set de ollas  Artículos para el hogar   151800.0          8300.0   
2357   Cuerda para saltar     Deportes y diversión    11200.0          1300.0   

     Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0         11/11/2022    Rafael Acosta          Bogotá             4   
1         03/01/2020  Izabela de León          Cúcuta             5   
2         25/12/2022    Maria Alfonso        Medellín             5   
3         23/11/2021    Lucas Olivera        Medellín             3   
4         25/04/2022   Blanca Ramirez          Bogotá             4   
...              ...              ...             ...           ...   
2353      12/02/2020    Camila Rivera        Medellín             5   
2354      17/05/2022    Felipe Santos     Bucaramanga             5   
2355      03/05/2022    Camila Rivera          Bogotá             5   
2356      15/06/2020   Blanca Ramirez            Cali             5   
2357      17/11/2021    Felipe Santos          Bogotá             5   

          Método de pago  Cantidad de cuotas      lat       lon  
0     Tarjeta de crédito                   3  4.60971 -74.08175  
1     Tarjeta de crédito                   2  7.89391 -72.50782  
2                  Nequi                   1  6.25184 -75.56359  
3                  Nequi                   1  6.25184 -75.56359  
4     Tarjeta de crédito                   7  4.60971 -74.08175  
...                  ...                 ...      ...       ...  
2353  Tarjeta de crédito                   1  6.25184 -75.56359  
2354  Tarjeta de crédito                   3  7.12539 -73.11980  
2355  Tarjeta de crédito                   1  4.60971 -74.08175  
2356  Tarjeta de crédito                   1  3.43722 -76.52250  
2357  Tarjeta de crédito                   1  4.60971 -74.08175  

[2358 rows x 12 columns]

In [6]:
facturacion_1 = df['Precio'].sum()
facturacion_2 = df2['Precio'].sum()
facturacion_3 = df3['Precio'].sum()
facturacion_4 = df4['Precio'].sum()

print("Tienda 1:", facturacion_1)
print("Tienda 2:", facturacion_2)
print("Tienda 3:", facturacion_3)
print("Tienda 4:", facturacion_4)


Tienda 1: 1150880400.0
Tienda 2: 1116343500.0
Tienda 3: 1098019600.0
Tienda 4: 1038375700.0


La tienda 1 tiene la mayor facturacion total, facturando 1150880400.0

In [8]:
df['Categoría del Producto'].value_counts()


Categoría del Producto
Muebles                    465
Electrónicos               448
Juguetes                   324
Electrodomésticos          312
Deportes y diversión       284
Instrumentos musicales     182
Libros                     173
Artículos para el hogar    171
Name: count, dtype: int64

El producto mas vendido de la tienda 1 es Muebles con 465 unidades vendidas

In [10]:
df2['Categoría del Producto'].value_counts()


Categoría del Producto
Muebles                    442
Electrónicos               422
Juguetes                   313
Electrodomésticos          305
Deportes y diversión       275
Instrumentos musicales     224
Libros                     197
Artículos para el hogar    181
Name: count, dtype: int64

In [ ]:
El producto mas vendido de la tienda 2 es Muebles con 422 unidades vendidas

In [11]:
df3['Categoría del Producto'].value_counts()

Categoría del Producto
Muebles                    499
Electrónicos               451
Juguetes                   315
Electrodomésticos          278
Deportes y diversión       277
Libros                     185
Instrumentos musicales     177
Artículos para el hogar    177
Name: count, dtype: int64

In [ ]:
El producto mas vendido de la tienda 3 es Muebles con 499 unidades vendidas

In [12]:
df4['Categoría del Producto'].value_counts()

Categoría del Producto
Muebles                    480
Electrónicos               451
Juguetes                   338
Deportes y diversión       277
Electrodomésticos          254
Artículos para el hogar    201
Libros                     187
Instrumentos musicales     170
Name: count, dtype: int64

El producto mas vendido de la tienda 4 es Muebles con 480 unidades vendidas

In [16]:
df['Producto'].value_counts().head(1)


Producto
Microondas    60
Name: count, dtype: int64

EN la tienda uno, el producto mas vendido es el Microondas con 60 unidades vendidas

In [15]:
df['Producto'].value_counts().tail(1)

Producto
Celular ABXY    33
Name: count, dtype: int64

En la teinda uno, el producto menos vendido es el Celular ABXY con 33 unidades vendidas

In [17]:
df2['Producto'].value_counts().head(1)

Producto
Iniciando en programación    65
Name: count, dtype: int64

En la tienda dos el producto mas vendido es Iniciando en Programacion con 65 unidades vendidas

In [19]:
df2['Producto'].value_counts().tail(1)

Producto
Juego de mesa    32
Name: count, dtype: int64

En la tienda dos el producto menos vendido es Juego de mesa con 32 unidades vendidas

In [20]:
df3['Producto'].value_counts().head(1)

Producto
Kit de bancas    57
Name: count, dtype: int64

En la tienda 3 el producto mas vendido es el Kit de bancas con 57 unidades vendidas

In [21]:
df3['Producto'].value_counts().tail(1)

Producto
Bloques de construcción    35
Name: count, dtype: int64

EN la tienda 3 el producto menos vendido es Bloques de construccion con 35 unidades vendidas

In [22]:
df4['Producto'].value_counts().head(1)

Producto
Cama box    62
Name: count, dtype: int64

En la tienda 4 el producto mas vendido es la Cama box con 62 unidades vendidas

In [23]:
df4['Producto'].value_counts().tail(1)

Producto
Guitarra eléctrica    33
Name: count, dtype: int64

En la Tienda 4 el producto menos vendido es la Guitarra Electrica con 33 unidades vendidas

In [25]:
df['Costo de envío'].mean()

np.float64(26018.60958033065)

A la tienda 1 le cuesta 26018.60 pesos hacer el envio

In [26]:
df2['Costo de envío'].mean()

np.float64(25216.235693090293)

A la tienda 2 le cuesta 25216.23 pesos hacer el envio

In [27]:
df3['Costo de envío'].mean()

np.float64(24805.680373039424)

A la tienda 3 le cuesta 24805.68 pesos hacer el envio

In [28]:
df4['Costo de envío'].mean()

np.float64(23459.457167090754)

A la tienda 4 le cuesta 23459.45 pesos hacer el envio

La tienda menos rentable es la Tienda 1 por que le cuesta 26018.60 pesos hacer el envio

Después de analizar la facturación total, las categorías más vendidas, los productos más y menos vendidos y el costo promedio de envío de las cuatro tiendas, se observa que la Tienda 1, a pesar de ser la que presenta la mayor facturación total con $1,150,880,400, también es la que tiene el costo promedio de envío más alto, con un valor de $26,018.60 por envío.

En cuanto a las categorías de productos, se identificó que Muebles es la categoría más vendida en todas las tiendas, lo que indica una alta competencia y poca diferenciación entre ellas. Además, aunque la Tienda 1 cuenta con productos con buenas ventas, como el Microondas, también presenta productos con menor rotación, como el Celular ABXY.

Considerando el elevado costo de envío en comparación con las demás tiendas y la similitud en las categorías más vendidas, se concluye que la Tienda 1 presenta oportunidades de optimización, por lo que se recomienda evaluar su venta para invertir en un nuevo negocio con menores costos operativos y mayor eficiencia logística.